packages/wallet/src/SingleAddressWallet.ts

```diff
@@ -446,6 +449,19 @@ export class SingleAddressWallet implements ObservableWallet {
     this.#logger.debug('Shutdown');
   }
 
+  async #getSignatures(txInternals: TxInternals, options?: SignTransactionOptions) {
+    let signatures: Cardano.Signatures = await this.keyAgent.signTransaction(txInternals, options);
+
+    if (options?.extraSigners) {
```

Comment:
Since KeyAgent takes in options that include the extraSigners, it would make sense for KeyAgents to call the extraSigners, or else KeyAgent.signTransaction has an option that's never used. Alternatively we could remove it from options used in KeyAgent.signTransaction and leave it a sign option only in scope of ObservableWallet/SingleAddressWallet

Reply:
addressed here https://github.com/input-output-hk/cardano-js-sdk/pull/404/commits/1745bf74c984767cac4ca69730a1da8f2ee40a90